app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -164,14 +176,36 @@ export default class List extends PureComponent {
     };
 
     buildSections = (props) => {
+        if (props.showLegacySidebar) {
+            const {
+                orderedChannelIds,
+            } = props;
+
+            return orderedChannelIds.map((s) => {
+                return {
+                    ...this.getSectionConfigByType(props, s.type),
+                    data: s.items,
+                };
+            });
+        }
+
         const {
+            channelsByCategory,
             orderedChannelIds,
         } = props;
 
-        return orderedChannelIds.map((s) => {
+        const unreads = orderedChannelIds.find((el) => el.type === 'unreads');
+
+        channelsByCategory.unshift({
+            channel_ids: unreads.items,
+            name: unreads.name,
+            type: unreads.type,
+        });
+
+        return channelsByCategory.map((cat) => {
             return {
-                ...this.getSectionConfigByType(props, s.type),
-                data: s.items,
+                ...this.getSectionConfigByType(props, cat.type, cat.display_name),
```

Comment:
`display_name` is also a pretty naive fix; I think the sectionlist/sectionHeader should pull this properly as props are already being passed into it.

Reply:
is display name really needed?